codec-http2/src/main/java/io/netty/handler/codec/http2/CompressorHttp2ConnectionEncoder.java

```diff
@@ -170,11 +170,11 @@ public ChannelFuture writeHeaders(final ChannelHandlerContext ctx, final int str
      * (alternatively, you can throw a {@link Http2Exception} to block unknown encoding).
      * @throws Http2Exception If the specified encoding is not not supported and warrants an exception
      */
-    protected EmbeddedChannel newContentCompressor(AsciiString contentEncoding) throws Http2Exception {
-        if (GZIP.equalsIgnoreCase(contentEncoding) || X_GZIP.equalsIgnoreCase(contentEncoding)) {
+    protected EmbeddedChannel newContentCompressor(ByteString contentEncoding) throws Http2Exception {
+        if (GZIP.equals(contentEncoding) || X_GZIP.equals(contentEncoding)) {
```

Comment:
Note this change.  `equalsIgnoreCase` doesn't fit well when dealing with binary strings.  You have to think about encodings and translations...blah.  HTTP/2 requires that header names are lower case.  So a byte wise comparison should be sufficient, unless the user switches off the "force to lower case default"...in which case they are explicitly taking this risk.